File: frontend/src/api/jobSla.ts
import request from './request'

export interface JobSla {
  id?: number
  jobName: string
  scheduleName?: string
  expectedDurationSec: number
  deviationPercent?: number
  enabled?: boolean
}

/** 最近执行对比行（QSYS2.JOB_LOG_INFO 大写列名） */
export interface SlaExecution {
  JOB_NAME: string
  SCHEDULE_NAME: string
  EXPECTED_SEC: number
  ACTUAL_SEC: number
  STATUS: 'OK' | 'BREACHED'
  [key: string]: unknown
}

export const fetchSlaRules = (): Promise<JobSla[]> => request.get('/job-sla/rules')

export const createSlaRule = (data: JobSla) => request.post('/job-sla/rules', data)

export const updateSlaRule = (id: number, data: JobSla) => request.put(`/job-sla/rules/${id}`, data)

export const deleteSlaRule = (id: number) => request.delete(`/job-sla/rules/${id}`)

/** 最近执行对比（Mock 仿真 / JT400 真实作业历史 QSYS2.JOB_LOG_INFO） */
export const fetchSlaExecutions = (): Promise<SlaExecution[]> => request.get('/job-sla/executions')

File: backend/rxas400adm-as400/src/main/java/com/rxas400adm/as400/entity/JobSla.java
package com.rxas400adm.as400.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 作业 SLA 规则（rx_job_sla）：为关键作业定义预期耗时，配合最近执行对比展示 SLA 达成情况。
 * 执行数据来自 AS400Client.jobSlaExecutions()（Mock 仿真 / JT400 真实作业历史 QSYS2.JOB_LOG_INFO）。
 */
@Data
@TableName("rx_job_sla")
public class JobSla {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 作业名（如 ORDNIGHT） */
    private String jobName;

    /** 作业说明/调度名 */
    private String scheduleName;

    /** 预期耗时（秒） */
    private Integer expectedDurationSec;

    /** 允许偏差百分比（超过即视为 BREACHED） */
    private Integer deviationPercent;

    private Boolean enabled;

    private LocalDateTime createdTime;

    private LocalDateTime updatedTime;
}
